packages/common/src/type-helpers.ts

```diff
@@ -61,3 +61,10 @@ export function errorCode(error: unknown): string | undefined {
 
   return undefined;
 }
+
+/**
+ * Asserts that some type is the never type
+ */
+export function assertNever(x: never): never {
```

Comment:
Make a specific helper for this use case and throw a TypeError.

Reply:
What do you mean specific helper? This will work for all cases where you want to do exhausive checks, pretty useful. The throw is actually irrelevant, since what matters happens at compile time, but I changed it anyway.

Reply:
Any extension of the enum will render this as a compile error. The throw could only happen I believe if you were doing weird JS things and going around TS.

Reply:
No, the server may be updated with new enum values and the SDK may not have those.

Reply:
Sure, makes sense.

Reply:
I guess my only request would be to pass in the error message to this function so it doesn't just say `Unexpected object` which isn't very helpful especially with enums.